### DISCUSSION TITLE
Filter status reset feature added - testing needed

### DISCUSSION THREAD

**Discussion by adorobis** [adorobis/hacomfoairmqtt#34] [Announcements]
Hello there,
@mweimerskirch @TimWeyand @PimDoos @mogwai2 

I have added new feature to read filter hours (I guess it is running hours since last reset), read and set filter weeks (I guess it is time after which filter status changes to Full) and a button to reset the filter status.
In general all three work for my system (ComfoAir 350), the filter status gets nicely reset (you need to power off the ventilation unit and power it back on after pressing the button - the same behavior as with standard control panels).
Unfortunately in my case the filter hours do not get reset and after a while (seems that after the filter hours counter increase) the filter status is back to Full.
I would appreciate if you could also test it and let me know if you observe similar behavior.

**Comment by PimDoos** [comment]
Got the latest version running on my Pi. So far it looks like it's all working fine, and the new entities show up nicely in the device view in Home Assistant ;)
I'll let it run for a few days and test the reset option. I'll let you know the results in a couple of days.
![image](https://user-images.githubusercontent.com/17709721/150656919-eec69770-34b5-4b2e-a5db-1879c06b526c.png)
![image](https://user-images.githubusercontent.com/17709721/150656923-a413d63c-cde9-4ed9-9120-923e5b87b414.png)

**Reply by adorobis** [reply]
Great! Could you please try to press the Filter Reset button after this turn of power supply from the unit, turn it back on and see if the Filter Hours gets reset to 0?

**Reply by PimDoos** [reply]
When the filter hours reach the set value, it seems to jump to a maximum value:
![image](https://user-images.githubusercontent.com/17709721/151424420-1a9a66d4-4bc7-4dc0-9b13-4732e781096e.png)

Before pressing the filter reset button:
![image](https://user-images.githubusercontent.com/17709721/151424439-edeacb3f-3a17-4051-83cd-bec927599ad4.png)

After pressing the filter reset button: (notice the hour value changed to another value)
![image](https://user-images.githubusercontent.com/17709721/151424748-f7065eae-afa8-45e5-bce3-3fb48471242e.png)

After power cycling the WHR930, the values are still the same as the last picture.

**Reply by adorobis** [reply]
Looks like some devices don't respond properly to this command. Following some advice from similar openhab integration author I've changed it so it now simulates button press to reset the filter. You can test the latest version. It works for me as expected now.

**Reply by adorobis** [reply]
Btw, looking at the number that you are receiving I have a feeling that it is not the actual number of filter hours as it should be a full number (integer) rather than a decimal. Or is the `.` in your setting meaning a thousands separator?

**Reply by PimDoos** [reply]
The `.` is a thousands seperator indeed (EU Country). Looks like this is some sort of binary command:
```
52744 = 0xCE08
52736 = 0xCE00
```

An hour after I pressed reset, the value is now `1 h`

**Reply by adorobis** [reply]
> An hour after I pressed reset, the value is now `1 h`

So this looks good now. I've had over 30000 hours in my unit and is now also counting from 0. Let's see after several weeks :)
